pyxdsm/XDSM.py

```diff
@@ -341,8 +378,15 @@ def add_process(self, systems, arrow=True):
             If true, arrows will be added to the process lines to indicate the direction
             of the process flow.
         """
-        self.processes.append(systems)
-        self.process_arrows.append(arrow)
+        sys_faded = {}
+        for s in self.systems:
+            sys_faded[s.node_name] = s.faded
+        if (self.auto_fade["processes"] == "all") or (
+            self.auto_fade["processes"] == "connected" and any([sys_faded[s] for s in systems])
+        ):
+            faded = True
+            print([sys_faded[s] for s in systems])
+        self.processes.append(Process(systems, arrow, faded))
```

Comment:
This currently throws an error if any of the items in the process chain are not in `self.systems`. I don't think we should break this capability because sometimes you want to put a step in a process that is not on the diagonal. In fact, this is done in the [kitchen sink example](https://github.com/mdolab/pyXDSM/blob/695c98e7343b077aa04dfa4566246e600ce9c70d/examples/kitchen_sink.py#L76)

Reply:
Good catch, I pushed a fix. Maybe we want to add some fading to the kitchen sink example or a test or something.